radix-engine/src/transaction/transaction_executor.rs

```diff
@@ -144,65 +130,127 @@ impl ExecutionConfig {
             max_panic_message_size: MAX_PANIC_MESSAGE_SIZE,
             max_number_of_logs: MAX_NUMBER_OF_LOGS,
             max_number_of_events: MAX_NUMBER_OF_EVENTS,
-            max_per_function_royalty_in_xrd: Decimal::try_from(MAX_PER_FUNCTION_ROYALTY_IN_XRD)
-                .unwrap(),
         }
     }
+}
 
-    pub fn for_genesis_transaction(network_definition: NetworkDefinition) -> Self {
+impl LimitParameters {
+    pub fn for_genesis_transaction() -> Self {
         Self {
-            enabled_modules: EnabledModules::for_genesis_transaction(),
             max_heap_substate_total_bytes: 512 * 1024 * 1024,
             max_track_substate_total_bytes: 512 * 1024 * 1024,
             max_number_of_events: 1024 * 1024,
+            ..Self::default()
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct SystemOverrides {
+    pub disable_costing: bool,
+    pub disable_limits: bool,
+    pub disable_auth: bool,
+    pub network_definition: Option<NetworkDefinition>,
+    pub costing_parameters: Option<CostingParameters>,
+    pub limit_parameters: Option<LimitParameters>,
+}
+
+impl Default for SystemOverrides {
+    fn default() -> Self {
+        Self {
+            disable_costing: false,
+            disable_limits: false,
+            disable_auth: false,
+            network_definition: None,
+            costing_parameters: None,
+            limit_parameters: None,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ExecutionConfig {
+    // These parameters do not affect state execution but only affect side effects
+    pub enable_kernel_trace: bool,
+    pub enable_cost_breakdown: bool,
+    pub execution_trace: Option<usize>,
+
+    pub network_definition: NetworkDefinition,
```

Comment:
For historical transaction replay, we will always need to pass the network definition. Does it make sense to store network definition in system parameters? Now, we have two source of truth at execution time.

Reply:
Good point. This should actually be an `Option<NetworkDefinition>`. I was debating whether it should always be passed in or not but settled for option approach. Happy to discuss.